### PR TITLE
Modified sigmoid by the elementwise interface.

### PR DESCRIPTION
**Pull request by huangxu96**
### PR types

Performance optimization
### PR changes

OPs
### Describe

Use elementwise interface to optimize sigmoid

### CODE REVIEW

File: paddle/fluid/operators/sigmoid_cross_entropy_with_logits_op.cu [hunk 1]
```diff
@@ -42,26 +44,86 @@ static inline int NumBlocks(const int N) {
...
+    outs[0] = dx_data;
+    outs[1] = counts;
+    return outs;
+  }
+};
+
+template <typename T>
+struct DivFunctor {
+  const T *norm_;
+  HOSTDEVICE inline DivFunctor(const T *norm) : norm_(norm) {}
```
Comment (AnnaTrainingG) [inline paddle/fluid/operators/sigmoid_cross_entropy_with_logits_op.cu]: 这里也可以直接使用double存储div =  1/norm 然后在下面使用loss = loss * div

File: paddle/fluid/operators/sigmoid_cross_entropy_with_logits_op.cu [hunk 2]
```diff
@@ -157,22 +203,36 @@ class GPUSigmoidCrossEntropyWithLogitsGradKernel
...
-      Sum<T, kNumCUDAThreads><<<1, kNumCUDAThreads, 0, dev_ctx.stream()>>>(
-          counts, limit, static_cast<T>(1e-5), norm);
-      Div<T><<<blocks, threads, 0, dev_ctx.stream()>>>(dx_data, limit, norm);
+      T *counts = counts_tensor->mutable_data<T>(context.GetPlace());
+      Tensor *norm_tensor = new Tensor();
+      norm_tensor->mutable_data<T>(context.GetPlace(), sizeof(T));
+      const std::vector<int> reduce_dim = {1};
+      TensorReduceImpl<T, T, kps::AddFunctor, NonzeroFunctor<T>>(
+          context.cuda_device_context(), *counts_tensor, norm_tensor,
+          NonzeroFunctor<T>(), reduce_dim, dev_ctx.stream());
```
Comment (AnnaTrainingG) [inline paddle/fluid/operators/sigmoid_cross_entropy_with_logits_op.cu:232]: 1.这里为什么使用NonzeroFunctor ? NonzeroFunctor有类型转换， 使用identityFunctor比较合适

**Review by Liu-xiandong — APPROVED** [review]

**Review by AnnaTrainingG — APPROVED** [review]
LGTM

**Review by xingfeng01 — APPROVED** [review]
